templates/infrastructure/account/tfEnvironment/tfRegion/document_db/terragrunt.hcl

```diff
@@ -0,0 +1,89 @@
+include {
+  path = find_in_parent_folders()
+}
+
+locals {
+  # Automatically load environment-level variables
+  account_vars = read_terragrunt_config(find_in_parent_folders("account.hcl"))
+  environment_vars = read_terragrunt_config(find_in_parent_folders("env.hcl"))
+  region_vars = read_terragrunt_config(find_in_parent_folders("region.hcl"))
+
+  # Extract out common variables for reuse
+  map_users = local.account_vars.locals.map_users
+  map_accounts = local.account_vars.locals.map_accounts
+
+  env = local.environment_vars.locals.environment
+
+  region = local.region_vars.locals.aws_region
+}
+
+# Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
+# working directory, into a temporary folder, and execute your Terraform commands in that folder.
+terraform {
+  # the below config is an example of what the config should like
+  source = "github.com/argonautdev/tf-modules.git//modules/aws/document_db?ref={{.RefVersion}}"
+}
+
+dependency "vpc" {
+  config_path = "../vpc"
+   mock_outputs = {
+    vpc_id = "temporary-dummy-id",
+    public_subnets = ["temporary-dummy-subnet-1", "temporary-dummy-subnet-2"],
+    private_subnets = ["temporary-dummy-subnet-1", "temporary-dummy-subnet-2"],
+    database_subnets = ["temporary-dummy-subnet-1", "temporary-dummy-subnet-2"],
+    vpc_cidr_block = "10.0.0.0/8"
+    default_security_group_id = "default"
+  }
+}
+
+
+# These are the variables we have to pass in to use the module specified in the terragrunt configuration above
+inputs = {
+  default_tags = {
+    "argonaut.dev/name" = "{{.Spec.name}}"
+    "argonaut.dev/manager" = "argonaut.dev"
+    "argonaut.dev/type" = "DocumentDB"
+    "argonaut.dev/env/${local.env}" = "true"
+  }
+
+  aws_region = "${local.region}"
+
+  name = "{{.Spec.name}}"
+
+  {{if .Spec.zone_id}}zone_id= "{{.Spec.zone_id}}"{{end}}
+  {{if .Spec.allowed_security_groups}}allowed_security_groups={{.Spec.allowed_security_groups}}{{end}}
```

Comment:
Upon every apply, all the security groups in the vpc will be able to access the documentdb instance. Each time we create a new eks cluster we will need to do a re-apply for the documentdb, so the list of security-groups is updated and rules are configured accordingly.

The best way to do this would be to add all eks-clusters as dependency to document-db, this makes sure document-db will only be created after all the eks clusters have their security-group configured.

Reply:
Refer line 40. The `dependencies` block.

Reply:
Dependency to ecrs has been removed since we now use cidr for access